Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -619,3 +613,23 @@ void HexEditorWidget::drop_event(GUI::DropEvent& event)
         open_file(response.value());
     }
 }
+
+DeprecatedString HexEditorWidget::make_display_string(DeprecatedString string)
+{
+    StringBuilder builder;
+    for (auto code_point : string) {
```

Comment:
Missing here are at least bell \a, and most <64 ASCII control characters.

Reply:
This is true. I actually chose the characters I in `make_display_string` by seeing which ones were displayed literally. All the ones I left out are rendered with `<?>`, which I was fine with

I see now (after responding to your above comment), the interpretation of `<?>` is ambiguous. Up until this point, I was assuming it could either mean "invalid character" or "some ascii control character"

My goal with `make_display_string` was to narrowly fix the issue of having it be so control characters obfuscate later characters (like `\n`)

However, I realize now, the way we render various control characters could change tomorrow (they could start rendering not as `<?>`) and we would have to fix it, so it does seem like we should handle all the control characters

I'll have to think whether I'm willing to do that in this PR, or whether it's out of scope

Reply:
\a is not affected by this question so it should be included either way. There are proper replacement characters defined by Unicode in some higher code points which are supposed to be used for this purpose. However, if Serenity has a sane way of rendering them at the moment, I guess that's fine.

Reply:
So, I wound up removing display string logic. Reason being, I don't have a good sense for what we should actually do

I'm considering it outside the scope of this issue. I plan to create a new issue and create another PR that references it so if someone with a better sense can pick it up and run with it